lib/vertex_client/utils/adjustment_allocator.rb

```diff
@@ -47,7 +48,7 @@ def validate!
       end
 
       def weights_format_valid?
-        weights.all? { |weight| weight.is_a?(Numeric) && weight >= 0 && /^\d+(?:\.\d+)?$/.match(weight.to_s) } && !weights_total.zero?
+        weights.all? { |weight| weight.is_a?(Numeric) && weight >= 0 } && !weights_total.zero?
```

Comment:
super minor, but you could change the end of this to `... && weights_total.positive?`

Reply:
Scratch that. I had to pull it out because it [failed](https://github.com/customink/rails_frontend/pull/3433) on Ruby 2.1.10